lib/api/objectPutPart.js

```diff
@@ -250,7 +254,9 @@ function objectPutPart(authInfo, request, streamingV4Params, log,
         (destinationBucket, objectLocationConstraint, cipherBundle,
         partKey, prevObjectSize, oldLocations, partInfo, next) => {
             // NOTE: set oldLocations to null so we do not batchDelete for now
-            if (partInfo && partInfo.dataStoreType === 'azure') {
+            if (partInfo &&
+               (partInfo.dataStoreType === 'azure' ||
+                partInfo.dataStoreType === 'gcp')) {
```

Comment:
could use a similar structure as my previous comment, with the `Array.includes()` method